lib/core/topologies/replset_state.js

```diff
@@ -420,9 +420,7 @@ ReplSetState.prototype.update = function(server) {
   //
   if (!this.primary && ismaster.ismaster && ismaster.setName) {
     var ismasterElectionId = server.lastIsMaster().electionId;
-    this.setName = this.setName.trim();
-    ismaster.setName = ismaster.setName.trim();
-    if (this.setName && this.setName !== ismaster.setName) {
+    if (this.setName && this.setName != 'undefined' && this.setName.trim() !== ismaster.setName.trim()) {
```

Comment:
temporary fix that removes whitespace of `setName` when comparing `this` with `ismaster` but doesn't change the variable values.

Reply:
This is a fine temporary fix, but we need to figure out what is causing the unnecessary whitespace before this PR is approved.

Reply:
Has there been progress on figuring out the whitespace issue?

Reply:
No - I haven't found been able to find anywhere that would trim or modify the setName.